domain/domain_test.go

```diff
@@ -0,0 +1,105 @@
+package domain
+
+import (
+	"testing"
+)
+
+var targetTests = []struct {
+	in  string
+	out Target
+}{
+	{"+target", Target{Target: "target", LocalPath: "."}},
+	{"+another-target", Target{Target: "another-target", LocalPath: "."}},
+	{"./a/local/dir+target", Target{Target: "target", LocalPath: "./a/local/dir"}},
+	{"/abs/local/dir+target", Target{Target: "target", LocalPath: "/abs/local/dir"}},
+	{"../rel/local/dir+target", Target{Target: "target", LocalPath: "../rel/local/dir"}},
+	{"github.com/foo/bar+target", Target{Target: "target", Registry: "github.com", ProjectPath: "foo/bar"}},
+	{"github.com/foo/bar:tag+target", Target{Target: "target", Registry: "github.com", ProjectPath: "foo/bar", Tag: "tag"}},
+	{"github.com/foo/bar:tag/with/slash+target", Target{Target: "target", Registry: "github.com", ProjectPath: "foo/bar", Tag: "tag/with/slash"}},
+	// \+
+	{"./a/local/dir-with-\\+-in-it+target", Target{Target: "target", LocalPath: "./a/local/dir-with-+-in-it"}},
+	{"/abs/local/dir-with-\\+-in+target", Target{Target: "target", LocalPath: "/abs/local/dir-with-+-in"}},
+	{"../rel/local/dir-with-\\+-in+target", Target{Target: "target", LocalPath: "../rel/local/dir-with-+-in"}},
+	{"github.com/foo/bar/dir-with-\\+-in+target", Target{Target: "target", Registry: "github.com", ProjectPath: "foo/bar/dir-with-+-in"}},
+	{"github.com/foo/bar:tag-with-\\+-in+target", Target{Target: "target", Registry: "github.com", ProjectPath: "foo/bar", Tag: "tag-with-+-in"}},
+}
+
+func TestTargetParser(t *testing.T) {
+	for _, tt := range targetTests {
+		t.Run(tt.in, func(t *testing.T) {
+			out, err := ParseTarget(tt.in)
+			if err != nil {
```

Comment:
If you use the testify package, this can be shortened into:
```
	NoError(t, err, "ParseTarget failed")
	Equal(t, tt.out, out)
```
like what was done in https://github.com/earthly/earthly/blob/main/autocomplete/complete_test.go#L74-L75

Reply:
:+1: